File: Substitute_Algorithm.py
# After
def foundPerson(people):
    for i in range(len(people)):
        if people[i] == "Don":
            return "Don"
        if people[i] == "John":
            return "John"
        if people[i] == "Kent":
            return "Kent"
    return ""

# Refactoring
def foundPerson(people):
    candidates = ["Don", "John", "Kent"]
    for i in range(len(people)):
        if people[i] in candidates:
            return people[i]
    return ""

File: Substitute_Algorithm.js
// After
function foundPerson(people) {
    people.forEach(function(element, i, array) {
        if (people[i] == "Don") {
            return "Don";
        }    
        if (people[i] == "John") {
            return "John";
        }    
        if (people[i] == "Kent") {
            return "Kent";
        }    
    });        
    return "";
}    

// Refactoring
function foundPerson(people) {
   ["Don", "John", "Kent"].forEach(function(element, i, array) {
	  people.forEach(function(name, y, people) {
  		if (name == element) {
  			return name;
  		}
	  });
  });         
  return "";
}    
